### PR TITLE
Change over solar panel breakage to use more construction graphs

### PR DESCRIPTION
**Pull request by 20kdc**
## About the PR

Construction graphs 1, IBreakAct 0.

More or less see changelog.

**Changelog**

:cl:
- tweak: Broken solar panels now spit out shards when deconstructed.
- fix: Solar assemblies are now de-rotated when deconstructing.

### CODE REVIEW

File: Content.Server/Solar/Components/SolarPanelComponent.cs [hunk 1]
```diff
@@ -16,67 +15,33 @@ namespace Content.Server.Solar.Components
...
         public int MaxSupply
         {
             get => _maxSupply;
             set {
                 _maxSupply = value;
-                UpdateSupply();
+                // This should almost never be called, shush.
+                EntitySystem.Get<PowerSolarSystem>().UpdateSupply(OwnerUid, this);
             }
         }
```
Comment (PaulRitter) [inline Content.Server/Solar/Components/SolarPanelComponent.cs:37]: just make it a friend of the system & get rid of the property imo

**Review by PaulRitter — CHANGES_REQUESTED** [review]
otherwise lgtm

File: Content.Server/Solar/Components/SolarPanelComponent.cs (hunk 1 above)
Comment (20kdc) [reply Content.Server/Solar/Components/SolarPanelComponent.cs:37]: I knew someone was going to comment on this...
Look, if I do that, it can't be VV'd

**Comment by 20kdc** [comment]
Re-requested review because MaxSupply is no longer writable via VV, so there's no longer an UpdateSupply call there or any other logic.

**Review by PaulRitter — APPROVED** [review]

**Pull request merged**